### ISSUE TITLE
Magento_ProductAlert unsubscribe/stock controller does not accept GET requests

### ISSUE THREAD

**Issue by bneff84** [magento/magento2#28318]
https://github.com/magento/magento2/blob/56ae4d5bbf181c0b62aeddb893705f900a8d5cb1/app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php#L27

The productalert/unsubscribe/stock controller route does not accept GET requests despite the implementation within the Magento_ProductAlert module only using GET requests against the endpoint. This causes unsubscribe URLs to redirect to a 404 route and log the following to debug.log

`[2020-05-21 12:52:38] report.DEBUG: URI '/productalert/unsubscribe/stock/product/112/'' cannot be accessed with GET method (Magento\ProductAlert\Controller\Unsubscribe\Stock) [] []`

Seems this was added to the core in https://github.com/magento/magento2/pull/23459 despite this exact issue being documented by someone in the ticket: https://github.com/magento/magento2/pull/23459#issuecomment-507273975

The response from @yuriichayka states to test this using Postman or another means of submitting a POST request against the controller, yet the implementation of this in the Magento_ProductAlert module uses GET requests (direct links) as seen in https://github.com/magento/magento2/blob/7caf49278aa88391decae5d1a6a5afe4427e6ad9/app/code/Magento/ProductAlert/view/frontend/templates/email/stock.phtml#L38

I don't actually see anywhere in core code where this controller receives a POST request from the core codebase, but the simplest solution would be to allow both GET and POST requests by adding the `HttpGetActionInterface` to the `\Magento\ProductAlert\Controller\Unsubscribe\Stock` class as well.

The `\Magento\ProductAlert\Controller\Unsubscribe\Price` controller already implements the GET interface, so I think it would probably be best to keep both controllers in-line with each other. I would either make them both accept POST and GET or simply convert them both to GET only as this is the only need I can see at the moment.

```
Index: app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php	(date 1584987777421)
+++ app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php	(date 1584987777421)
@@ -11,6 +11,7 @@
 use Magento\Catalog\Api\ProductRepositoryInterface;
 use Magento\Customer\Model\Session as CustomerSession;
 use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\App\ObjectManager;
 use Magento\Framework\Controller\Result\Redirect;
@@ -24,7 +25,7 @@
 /**
  * Unsubscribing from 'back in stock alert'.
  */
-class Stock extends UnsubscribeController implements HttpPostActionInterface
+class Stock extends UnsubscribeController implements HttpPostActionInterface, HttpGetActionInterface
 {
     /**
      * @var ProductRepositoryInterface
```

**Comment by frostblogNet** [comment]
Preconditions (*)

Magento v2.3.5 p1

Steps to reproduce (*)

- Set a product to out of stock
- Enable Product allerts for back in stock
- Log in  to a frontend account
- go to the page of the product whith is out of stock
- click the link for product alerts
- run cron job catalog_product_alert
- open the product alert e-mail
- click on the unsubscribe link

Expected result (*)

- being unsubscribe from this product allert
- message that you have be unsubscribed

Actual result (*)

- getting a 404 page

@bneff84  patch file is correct to solve this issue. The url call form the mail is in http Get context.

**Comment by frostblogNet** [comment]
@mageto @magento-engcom-team  please review again

**Comment by cyildirim** [comment]
@yuriichayka Is there any specific reason that's implemented with `HttpPostActionInterface` ? Unsubscribe from email urls doesnt work as `\Magento\ProductAlert\Controller\Unsubscribe\Stock` only acceps post request.

**Comment by Hexmage** [comment]
Experiencing this as well. The links in the email are obviously GET, but the code expects a POST. Looks like this is still broken in the latest dev branch.

**Comment by cyildirim** [comment]
This ticket can be closed as this issue is fixed on `2.3.6-p1` with following commit  https://github.com/magento/magento2/commit/851501e9435c95bc361794901a0f135a8974b6e7

**Comment by eduard13** [comment]
Closing the issue. It seems to be fixed into the latest version https://github.com/magento/magento2/pull/32957#issuecomment-844120351
